media/client/ipc/source/MediaKeysCapabilitiesIpc.cpp

```diff
@@ -42,6 +43,8 @@ std::shared_ptr<IMediaKeysCapabilitiesIpcFactory> IMediaKeysCapabilitiesIpcFacto
 
 std::shared_ptr<IMediaKeysCapabilities> MediaKeysCapabilitiesIpcFactory::getMediaKeysCapabilitiesIpc() const
 {
+    std::lock_guard<std::mutex> lock{m_creationMutex};
```

Comment:
You won't need those creation mutexes if You use Meyer's singleton. It's generally thread-safe:
https://laristra.github.io/flecsi/src/developer-guide/patterns/meyers_singleton.html

Reply:
I like this but I don't think it's a priority right now. Let's deliver this as is if there are no other comments and I've raised https://jira.rdkcentral.com/jira/browse/RIALTO-115 to make this improvement.